scripts/referrer-user-count.ts

```diff
@@ -0,0 +1,100 @@
+import { stringify } from 'csv-stringify/sync'
+import { writeFileSync } from 'fs'
+import yargs from 'yargs'
+import { supportedNetworkIds } from './networks'
+import { protocolFilters } from './filters'
+import { fetchReferralEvents, removeDuplicates } from './referrals'
+import { NetworkId, Protocol, protocols } from './types'
+
+async function getArgs() {
+  const argv = await yargs
+    .env('')
+    .option('protocol', {
+      alias: 'p',
+      description: 'protocol that the referrals are for',
+      demandOption: true,
+      choices: protocols,
+      type: 'string',
+    })
+    .option('output-file', {
+      alias: 'o',
+      description: 'output file path to write JSON results',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('referrer-ids', {
+      alias: 'r',
+      description: 'a comma separated list of referrers IDs',
+      type: 'string',
+      demandOption: false,
+      coerce: (arg) => new Set(arg.split(',')),
+    })
+    .option('network-ids', {
+      alias: 'n',
+      description: 'Comma-separated list of network IDs',
+      type: 'string',
+      demandOption: false,
+      coerce: (arg: string) => new Set(arg.split(',')),
+    }).argv
+
+  return {
+    protocol: argv['protocol'] as Protocol,
+    protocolFilter: protocolFilters[argv['protocol'] as Protocol],
+    networkIds: (argv['network-ids'] as NetworkId) ?? supportedNetworkIds,
+    referrers: argv['referrer-ids'] as string,
```

Comment:
isn't this a set?

Reply:
I had done this to allow yargs to take arrays as `--referrer-ids val1,val2,val3` instead of `--referrer-ids val1 val2 val3`, but I am fine with either option. If we choose to use yargs array type we can cast these to `string[]`.

Reply:
Changed to arrays in dd0514e966c49cd9cd7c7a473e37e00f393c797e.